config/config.go

```diff
@@ -56,6 +56,9 @@ type Config struct {
 	NATManager  NATManagerC
 	Peerstore   pstore.Peerstore
 	Reporter    metrics.Reporter
+
+	PingCustom bool
+	Ping       bool
```

Comment:
Not sure I'm liking the proliferation of the `...Custom` fields. If we make the actual flags pointers `*bool`, a `nil` could indicate no user-provided value. WDYT? Same for Relay/RelayCustom?

Reply:
I tried with a ternary enum, but it was disgusting to work with :)

Reply:
How about modelling the ternary state like this? https://github.com/libp2p/go-libp2p/commit/4758a9552e20447bf34e7543d7607e7a5fb9ce76

Reply:
ugh, that's ugly too. I don't mind the Custom thingies that much, because the struct is only used for configuring the host and nothing else.

Reply:
Why not `DisablePing`?